include/aspect/postprocess/geoid.h

```diff
@@ -0,0 +1,347 @@
+/*
+  Copyright (C) 2011, 2012 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__postprocess_geoid_h
+#define __aspect__postprocess_geoid_h
+
+#include <aspect/simulator.h>
+#include <aspect/simulator_access.h>
+#include <aspect/postprocess/interface.h>
+#include <aspect/postprocess/boundary_pressures.h>
+#include <aspect/postprocess/boundary_densities.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    using namespace dealii;
+
+    namespace internal
+    {
+      /**
+       * A struct that contains a representation of the coefficients of a
+       * harmonic expansion.
+       */
+      template <int dim>
+      struct HarmonicCoefficients
+      {
+        HarmonicCoefficients(const unsigned int max_degree);
+
+        std::vector<double> sine_coefficients;
+        std::vector<double> cosine_coefficients;
+      };
+
+      /**
+       * Abstract base struct to expand arbitrary fields of doubles into multipole
+       * moments.  In 3D this is spherical multipole moments, and in
+       * 2D it is cylindrical multipole moments.
+       */
+      template <int dim>
+      struct MultipoleExpansion
+      {
+        MultipoleExpansion(const unsigned int max_degree);
+
+        /**
+         *Do the multipole expansion at a particular quadrature point.
+         *
+         *@param position The location of the quadrature point.
+         *
```

Comment:
I think usually we put no empty line between parameter documentation. I found some places, where we do, but it looks a bit separated here, so please remove the empty lines